hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c

```diff
@@ -0,0 +1,231 @@
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Sensirion AG nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "sensirion_i2c.h"
+#include "sensirion_common.h"
+
+#include "cy_pdl.h"
+#include "cycfg.h"
+
+
+
+/*******************************************************************************
+ * global variables
+ *******************************************************************************/
+static cy_stc_scb_i2c_context_t mI2C_context;
+
+/* Combine master error statuses in single mask  */
+#define MASTER_ERROR_MASK  (CY_SCB_I2C_MASTER_DATA_NAK | CY_SCB_I2C_MASTER_ADDR_NAK    | \
+                            CY_SCB_I2C_MASTER_ARB_LOST | CY_SCB_I2C_MASTER_ABORT_START | \
+                            CY_SCB_I2C_MASTER_BUS_ERR)
+
+/*******************************************************************************
+* Forward declaration
+*******************************************************************************/
+void mI2C_Interrupt(void);
+
+
+void mI2C_Interrupt(void)
+{
+    Cy_SCB_I2C_Interrupt(mI2C_HW, &mI2C_context);
+}
+
+/* INSTRUCTIONS
+ * ============
+ *
+ * Implement all functions where they are marked as IMPLEMENT.
+ * Follow the function specification in the comments.
+ */
+
+/**
+ * Initialize all hard- and software components that are needed for the I2C
+ * communication.
+ */
+void sensirion_i2c_init()
+{
+	cy_en_scb_i2c_status_t initStatus;
+	cy_en_sysint_status_t sysStatus;
+	cy_stc_sysint_t mI2C_SCB_IRQ_cfg =
+	{
+			/*.intrSrc =*/ scb_3_interrupt_IRQn,
+			/*.intrPriority =*/ 2UL
+	};
+	/* Initialize and enable the I2C in master mode. */
+	initStatus = Cy_SCB_I2C_Init(mI2C_HW, &mI2C_config, &mI2C_context);
+	if(initStatus != CY_SCB_I2C_SUCCESS)
+	{
+		// return STATUS_FAIL;
```

Comment:
@abrauchli We should really change the interface to add error codes for the `sensirion_i2c_init` function.

Reply:
yes, let's